### PR TITLE
Re-added Files from Other Project

### PR DESCRIPTION
**Pull request by svhnstorage**
Since we started a project from scratch, I had to re add all amplify related files and the readme. A build error popped up with styles that XML that was temporarily patched out via comment. AuthManager added, login/logout have UI functionality. Next to be added sign up UI functionality, styling and UX quality.

dired

### CODE REVIEW

**Comment by shikdernyc** [comment]
Can you add gif and screenshot of the UI

File: app/src/main/java/com/example/mealsnap_prototype_2/ConfigActivity.java [hunk 1]
```diff
@@ -2,13 +2,69 @@
...
+        try {
+            AmplifyService.configure(getApplicationContext());
+            Log.i("MyAmplifyApp", "Initialized AmplifyService");
+        } catch (AmplifyException error) {
+            Log.e("MyAmplifyApp", "Could not initialize AmplifyService", error);
+        }
+
+        //AuthHandler.signUp("svhnstorage@gmail.com", "heygirl", "Heygirl123~");
+
+        AuthHandler.signOut(new AuthEvents() {
```
Comment (shikdernyc) [inline app/src/main/java/com/example/mealsnap_prototype_2/ConfigActivity.java]: Remove this.

File: app/src/main/java/com/example/mealsnap_prototype_2/LoginActivity.java [hunk 2]
```diff
@@ -30,7 +40,44 @@ protected void onCreate(Bundle savedInstanceState) {
 
         tvLogin.setText("Welcome Back");
 
-        btnLogin.setOnClickListener(v -> goToMainActivity());
+        //TODO Create button states with greyed out if all fields are not filled in
+        btnLogin.setOnClickListener(v -> AuthHandler.signOut(new AuthEvents() {
```
Comment (shikdernyc) [inline app/src/main/java/com/example/mealsnap_prototype_2/LoginActivity.java]: Can you move the signout logic inside of AuthHandler.login function

File: app/src/main/java/com/example/mealsnap_prototype_2/LoginActivity.java [hunk 3]
```diff
@@ -30,7 +40,44 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                        goToMainActivity();
+                    }
+
+                    @Override
+                    public void onFailure(AuthException authError) {
+                        Log.i(TAG,"Sign in failed");
+                        runOnUiThread(new Runnable() {
+                            @Override
+                            public void run() {
+                                Toast.makeText(LoginActivity.this, "Username or Password Didn't Match", Toast.LENGTH_SHORT).show();
```
Comment (shikdernyc) [inline app/src/main/java/com/example/mealsnap_prototype_2/LoginActivity.java:73]: This is not necessarily always the same error. There should be an error message as part of the `authError` object. Try displaying that instead.

File: app/src/main/java/com/example/mealsnap_prototype_2/LoginActivity.java [hunk 4]
```diff
@@ -30,7 +40,44 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                        });
+                        //TODO Error Handling
+                    }
+                });
+            }
+
+            @Override
+            public void onFailure(AuthException authError) {
+                Log.i(TAG, "Outter signout failed");
+                //TODO Error Handling
```
Comment (shikdernyc) [inline app/src/main/java/com/example/mealsnap_prototype_2/LoginActivity.java:84]: Are you handling this error on a separate PR?

File: app/src/main/java/com/example/mealsnap_prototype_2/SettingsActivity.java [hunk 5]
```diff
@@ -20,7 +29,27 @@ protected void onCreate(Bundle savedInstanceState) {
 
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
 
-        btnLogout.setOnClickListener(v -> goBackToLogin());
+        btnLogout.setOnClickListener(v -> {
+            AuthHandler.signOut(new AuthEvents() {
```
Comment (shikdernyc) [inline app/src/main/java/com/example/mealsnap_prototype_2/SettingsActivity.java]: I would extract the logic that handles logging out to its on function. For example, create another function in `SettingsActivity` called `handleLogout()`, which would run the logout logic. the on click listener would then be
```
btnLogout.setOnClickListener(v -> { this.handleLogout() } )
```

I would also do this for login and other methods as well. Keeping all these logic inside of the onCreate function makes the function harder to read. Especially as we start having more and more listeners for other buttons

File: app/src/main/java/services/authhandler/AuthHandler.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import android.util.Log;
+
+import com.amplifyframework.auth.AuthUserAttributeKey;
+import com.amplifyframework.auth.options.AuthSignUpOptions;
+import com.amplifyframework.core.Amplify;
+
+public class AuthHandler{
+
+    public static void signUp(String email, String username, String password, AuthEvents eventStatus){
+        AuthSignUpOptions options = AuthSignUpOptions.builder()
```
Comment (shikdernyc) [inline app/src/main/java/services/authhandler/AuthHandler.java:14]: Fix indentations

File: app/src/main/java/services/authhandler/AuthHandler.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+                .userAttribute(AuthUserAttributeKey.email(), email)
+                .build();
+        Amplify.Auth.signUp(username, password, options,
+                result -> {
+                eventStatus.onSuccess();
+                Log.i("AuthQuickStart", "Result: " + result.toString());
```
Comment (shikdernyc) [inline app/src/main/java/services/authhandler/AuthHandler.java]: Extract "AuthQuickStart" to a `TAG` variable

**Review by shikdernyc — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/mealsnap_prototype_2/ConfigActivity.java (hunk 1 above)
Comment (svhnstorage) [reply app/src/main/java/com/example/mealsnap_prototype_2/ConfigActivity.java]: Done

File: app/src/main/java/services/authhandler/AuthHandler.java (hunk 6 above)
Comment (svhnstorage) [reply app/src/main/java/services/authhandler/AuthHandler.java:14]: This the default, as well as auto code reformat, indentation cause they are parameters to a function.

File: app/src/main/java/services/authhandler/AuthHandler.java (hunk 7 above)
Comment (svhnstorage) [reply app/src/main/java/services/authhandler/AuthHandler.java]: Done

File: app/src/main/java/services/authhandler/AuthHandler.java (hunk 6 above)
Comment (svhnstorage) [reply app/src/main/java/services/authhandler/AuthHandler.java:14]: Tweaked it anyways

File: app/src/main/java/com/example/mealsnap_prototype_2/LoginActivity.java (hunk 4 above)
Comment (svhnstorage) [reply app/src/main/java/com/example/mealsnap_prototype_2/LoginActivity.java:84]: Yes, will not be done in this PR

File: app/src/main/java/com/example/mealsnap_prototype_2/LoginActivity.java (hunk 3 above)
Comment (svhnstorage) [reply app/src/main/java/com/example/mealsnap_prototype_2/LoginActivity.java:73]: Errors will be better handled in a separate PR

File: app/src/main/java/com/example/mealsnap_prototype_2/SettingsActivity.java (hunk 5 above)
Comment (svhnstorage) [reply app/src/main/java/com/example/mealsnap_prototype_2/SettingsActivity.java]: Done

File: app/src/main/java/com/example/mealsnap_prototype_2/LoginActivity.java (hunk 2 above)
Comment (svhnstorage) [reply app/src/main/java/com/example/mealsnap_prototype_2/LoginActivity.java]: Done

**Review by qiaoganchen22 — APPROVED** [review]

**Pull request merged**